cloudhealth/api.go

```diff
@@ -0,0 +1,85 @@
+package cloudhealth
+
+import (
+	"time"
+)
+
+// Variable represents a cloudfire variable rolled up over some amount of time
+type Variable struct {
+	Count uint64
+	Min   float64
+	Max   float64
+	Sum   float64
+}
+
+// IsEmpty returns true if no values have been added to this variable
+func (v *Variable) IsEmpty() bool {
+	return v.Count == 0
+}
+
+// Add records a new value in this variable
+func (v *Variable) Add(x float64) {
+}
+
+// Avg returns the average of this variable
+func (v *Variable) Avg() float64 {
+	return v.Sum / float64(v.Count)
+}
+
+const InstanceDataPointCount = 12 // 4 variables * (min,max,avg)
+
+// InsstanceData contains rolled up data for a particular instance
+type InstanceData struct {
+	// The aws instance ID
+	InstanceId string
+	// The timestamp at one hour granularity
```

Comment:
Minor style nit: I find it more compact and clearer to have the comment in the same line as the field (as long as there's room). Otherwise I have to think whether the comment applies to the field line above or below.